app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -138,23 +187,62 @@ def process_availability(availability)
       end
 
       def parse_deployment(server, deployment)
-        {
+        specific = {
           :name              => parse_deployment_name(deployment.id),
-          :middleware_server => server, # TODO: does that make sense? What is better?
-          :nativeid          => deployment.id,
-          :ems_ref           => deployment.path
+          :middleware_server => server,
         }
+        parse_base_item(deployment).merge(specific)
       end
 
       def parse_datasource(server, datasource, config)
-        data = {
+        specific = {
           :name              => datasource.name,
           :middleware_server => server,
-          :nativeid          => datasource.id,
-          :ems_ref           => datasource.path
         }
         if !config.empty? && !config['value'].empty? && config['value'].respond_to?(:except)
-          data[:properties] = config['value'].except('Username', 'Password')
+          specific[:properties] = config['value'].except('Username', 'Password')
+        end
+        parse_base_item(datasource).merge(specific)
+      end
+
+      def parse_middleware_domain(domain)
+        specific = {
+          :name      => domain.properties['Name'],
+          :type_path => domain.type_path,
+        }
+        parse_base_item(domain).merge(specific)
+      end
+
+      def parse_middleware_server_group(group)
+        specific = {
+          :name      => parse_server_group_name(group.name),
+          :type_path => group.type_path,
+          :profile   => group.properties['Profile'],
+        }
+        parse_base_item(group).merge(specific)
+      end
+
+      def parse_middleware_server(eap, name = nil)
+        specific = {
+          :name      => name || parse_standalone_server_name(eap.id),
+          :type_path => eap.type_path,
+          :hostname  => eap.properties['Hostname'],
+          :product   => eap.properties['Product Name'],
+        }
+        parse_base_item(eap).merge(specific)
+      end
+
+      private
+
+      def parse_base_item(item)
```

Comment:
Yeah, it's too specific. Almost each entity has it's own parse method for the name. Because the default name that is reported by the agent can't be shown to mere mortals behind the UI. 

By using those `parse_foo_name` methods we basically create a tight coupling between the format of the name that agent reports (and what can possibly change) and the MiQ. In the ideal world the agent would already report some useful string that can be presented without any data mangling.